### ISSUE TITLE
Memory Buffer Compare

### ISSUE THREAD

**Issue by basvodde** [cpputest/cpputest#30]
Having a good memory buffer compare would probably be a very useful addition.

One version of this was written by Andreas Stenius:
https://git.astekk.se/pub/pub/cpputest/commit/?h=pub&id=7a0b85e205a03a0e87d9c33ccf9d0ff1bfce5f20

and:
https://git.astekk.se/pub/pub/cpputest/commit/?h=pub&id=3178e699147ab7a1e15f7012829cacada0c3b341

**Comment by KisImre** [comment]
The MEMCMP_EQUAL assertion is the part of the framework now, but the mock extension still not has a function for passing input arrays. It should not do anything but copy the contents of the input buffer into the buffer that we specified when we set our expectations. So basically it should do the same as the "withOutputParameterReturning" does but in the opposite direction. We will be able to check the contents of the buffer using the MEMCMP_EQUAL assertion. I would call this function "withInputParameterForwarding". I'm getting familiar with source code of the mock extension, and if you accept this concept I could implement it.
